utils/python/CIME/hist_utils.py

```diff
@@ -282,9 +301,23 @@ def generate_baseline(case, baseline_dir=None):
             if os.path.exists(baseline):
                 os.remove(baseline)
 
+            #special care for multi-instance cases,
+            #only keep first instance and
```

Comment:
> I think that a multi-instance test with different results in each instance is a fundamentally different test than what we have now. In the current tests we now allow for any number of ensemble members and expect them all to give the same result. Putting all of them in the baseline dir is a waste of space. This is also the logic by which the NCK test can compare the single instance output to the multi-instance output.

With respect to NCK: Yes, I agree that, for this particular test, the point is that the two instances are identical so there's no need to save baselines for both. However, we discussed quite a while ago (at a CSEG meeting, I believe) that really the NCK test is a pretty weak test of the multi-instance capabilities, and we really want a stronger test, where the instances differ in some way (e.g., via some namelist parameter). For example, I could imagine a test where you do 3 runs: (a) single instance with parameter foo = 17; (b) single instance with parameter foo = 42; (c) multi-instance with instance 1 having foo=17 and instance 2 having foo=42. Then do the appropriate comparisons. Nobody ever implemented something like this, even though there was general agreement that we should. I'd like to think that it's more likely to be implemented now that it's easier to write new tests.

But even now, outside the NCK test and its variants (NC*), there is no assumption (as far as I know) that the two instances of a multi-instance test are identical: If I run an SMS_N2 test, I can point to a testmods directory that has different test mods for inst 1 vs inst 2. Again, this would provide a much stronger test of multi-instance than the current typical practice. And for this to be a good test, I'd want baselines for both instances.

All that said: I realize that there are no tests like these right now, so if you'd rather have code to reduce disk usage now, knowing that we'll likely want to delete this code later, then I'm okay with that.